glyphs-reader/src/font.rs

```diff
@@ -264,3 +294,296 @@ impl Path {
+    match weight {
+        100 => "Thing",
+        200 => "Extra-light",
+        300 => "Light",
+        400 => "Regular",
+        500 => "Medium",
+        600 => "Semi-bold",
+        700 => "Bold",
+        800 => "Extra-bold",
+        900 => "Black",
+        _ => panic!("No name for weight {}", weight),
+    }
+}
+
+impl Font {
+    fn is_v2(&self) -> bool {
+        let mut is_v2 = true;
+        if let Some(Plist::Integer(version)) = self.other_stuff.get(".formatVersion") {
+            is_v2 = *version < 3; // .formatVersion is only present for v3+
+        }
+        is_v2
+    }
+
+    fn v2_to_v3_axes(&mut self) -> Result<Vec<String>, Error> {
+        let mut tags = Vec::new();
+        if let Some(Plist::Array(custom_params)) = self.other_stuff.get_mut("customParameters") {
+            // Axes migrates from customParameters
+            let mut axes_idx = None;
+            for (idx, custom_param) in custom_params.iter_mut().enumerate() {
+                if let Plist::Dictionary(dict) = custom_param {
+                    if Some(&Plist::String("Axes".to_string())) == dict.get("name") {
+                        axes_idx = Some(idx);
+
+                        if self.axes.is_none() {
+                            self.axes = Some(Vec::new());
+                        }
+                        let v3_axes = self.axes.as_mut().unwrap();
+                        let Some(Plist::Array(v2_axes)) = dict.get_mut("value") else {
+                            return Err(Error::StructuralError("No value for Axes custom parameter".into()));
+                        };
+                        for v2_axis in v2_axes {
+                            let Plist::Dictionary(v2_axis) = v2_axis else {
+                                return Err(Error::StructuralError("Axis value must be a dict".into()));
+                            };
+                            let tag = v2_axis.get("Tag").unwrap().as_str().unwrap();
+                            tags.push(tag.to_string()); 
+                            v3_axes.push(Axis {
+                                name: v2_axis.get("Name").unwrap().as_str().unwrap().into(),
+                                tag: tag.into(),
+                            });
+                        }
+                    }
+                }
+            }
+            axes_idx.map(|idx| custom_params.remove(idx));
+
+            if custom_params.is_empty() {
+                self.other_stuff.remove("customParameters");
+            }            
+        }
+        Ok(tags)
+    }
+
+    fn v2_to_v3_metrics(&mut self) -> Result<(), Error> {
+        // metrics are in parallel arrays in v3
+        let v3_metric_names = ["ascender".to_string(), "baseline".to_string(), "descender".to_string(), "cap height".to_string(), "x-height".to_string()];
+        let v2_metric_names = ["ascender".to_string(), "baseline".to_string(), "descender".to_string(), "capHeight".to_string(), "xHeight".to_string()];
```

Comment:
maybe these two arrays of strings for the metric names can be constant defined at the top

Reply:
Good idea, done. Ran afoul of the traditional no global String's but still an improvement I think.